rudderlabs/rudder-sdk-kotlin
# feat: remove anonymousId field from root.traits in group event

## Description


- To avoid setting the same `anonymousId` value multiple times, we are deviating away from the legacy SDK behaviour.
- Now this value will not be set for group event root.traits.
- Only user passed group traits will be used.

## Type of change


## Implementation Details


- We have removed setting of the anonymousId from group event root.traits.

## Checklist


## How to test?


Make a group event with traits and observe that anonymousId value is missing from `root.traits`.

```
{
    "anonymousId": "42ca997a-a933-4337-bdd4-442db683911d",
    "channel": "mobile",
    "context": {
        "app": {
            "build": "100",
            "name": "Rudder-Android-Libs",
            "namespace": "com.rudderstack.android.sampleapp",
            "version": "0.1.0"
        },
        "device": {
            "adTrackingEnabled": true,
            "advertisingId": "e4c2e606-8970-488b-8958-60503956d869",
            "id": "11c89c643bfb7b71",
            "manufacturer": "Google",
            "model": "sdk_gphone64_arm64",
            "name": "emu64a",
            "type": "Android"
        },
        "library": {
            "name": "com.rudderstack.android.sdk",
            "version": "1.0.0"
        },
        "locale": "en-US",
        "network": {
            "bluetooth": true,
            "carrier": "T-Mobile",
            "cellular": true,
            "wifi": true
        },
        "os": {
            "name": "Android",
            "version": "14"
        },
        "screen": {
            "density": 560,
            "height": 2808,
            "width": 1440
        },
        "sessionId": 1732205428,
        "sessionStart": true,
        "timezone": "Asia/Kolkata",
        "traits": {
            "anonymousId": "42ca997a-a933-4337-bdd4-442db683911d"
        }
    },
    "groupId": "Group at - New",
    "integrations": {
        "All": true
    },
    "messageId": "8309613b-2336-4c2e-bcf8-97ac5a7f1d40",
    "originalTimestamp": "2024-11-21T16:10:31.441Z",
    "receivedAt": "2024-11-21T16:10:33.054Z",
    "request_ip": "49.37.35.149",
    "rudderId": "7479ead0-88a2-4fc0-beb6-971ddf65ab03",
    "sentAt": "2024-11-21T16:10:32.804Z",
    "traits": {                                                                                                       <--- Note: AnonymousId is not set
        "key-1": "value-1"
    },
    "type": "group"
}
```

## Breaking Changes


- Previously, we set the `anonymousId` by default for group events in `root.traits`.
- From now, on we will not set this value, as this value already exists at multiple places (i.e., root.anonymousId and context.traits.anonymousId).

## Maintainers Checklist


## Screenshots (if applicable)


<img width="240" alt="image" src="https://github.com/user-attachments/assets/3b960bac-05c4-411a-b8e2-c25fcd586aa8">

## Additional Context
